### PR TITLE
integrated suggestions

### PR DESCRIPTION
**Pull request by JonathanFenske**
We updated the train and featurization scripts according to your remarks. We only did not integrate the shuffle() function yet because we need to talk about the details what  is supposed to happen.

### CODE REVIEW

File: src/featurization.py [hunk 1]
```diff
@@ -179,9 +207,17 @@ def getAvgFeatureVecs(reviews, model, num_features):
             clean_test_reviews=review_wordlist(review, remove_stopwords=True)
             testDataVecs=featureVecMethod(clean_test_reviews, model, num_features)
             writer.writerow(testDataVecs)
-            
-   
-    
+
+    # initializing the memory maps
+    print('creating memory map...')
+    trainXfile = csv.reader(open('data/features/80trainDataVec.csv', 'rt'))
```
Comment (DennisHamann) [inline src/featurization.py:213]: Dateiname auch von vecsize abhängig. model.vector_size kann genutzt werden

File: src/train.py [hunk 2]
```diff
@@ -44,12 +35,55 @@ def get_model(input_shape,output_shape):
...
-    #train model with given dataset
     return model
 
-#with open(output, 'w') as f:
-#    f.write('This is a trained model')
-
+def get_data(no_of_reviews_train,no_of_reviews_test,maxlen_train,maxlen_test,vecsize):
+    trainX = np.memmap('data/features/trainmapX', dtype='float', mode='r',shape=(no_of_reviews_train,maxlen_train,vecsize))
+    testX = np.memmap('data/features/testmapX', dtype='float', mode='r',shape=(no_of_reviews_test,maxlen_test,vecsize))
+    trainYfile = open('data/features/80trainDatalabel.pickle', 'rb')
```
Comment (DennisHamann) [inline src/train.py:43]: Dateiname von vecsize abhängig

File: src/train.py [hunk 3]
```diff
@@ -44,12 +35,55 @@ def get_model(input_shape,output_shape):
...
+    testYfile = open('data/features/80testDatalabel.pickle', 'rb')
+    trainY = pickle.load(trainYfile)
+    testY = pickle.load(testYfile)
+    return trainX, testX, trainY, testY
+
+def data2memmap(file,mmap,no_of_reviews,maxlen,vecsize):
```
Comment (DennisHamann) [inline src/train.py:49]: diese Funktion ist schon in featurization.py und kann hier gelöscht werden.

File: src/train.py [hunk 4]
```diff
@@ -44,12 +35,55 @@ def get_model(input_shape,output_shape):
...
+        review = list()
+        for i in range(review_length-1):
+            wordtmp = row[i].replace('     ', ' ').replace('    ', ' ').replace('   ', ' ').replace('  ', ' ').replace(' ]', '').replace(']', '').replace('[ ', '').replace('[', '').replace('\n', '')
+            wordtmp = wordtmp.split(' ')
+            review.append(np.array(wordtmp).astype('float'))
+        review = np.array(review)
+        data[idx, (maxlen - review.shape[0]):maxlen, :] = review
+    data.flush()
+
+def get_shape(file):
```
Comment (DennisHamann) [inline src/train.py:66]: diese Funktion ist schon in featurization.py und kann hier gelöscht werden.

File: src/train.py [hunk 5]
```diff
@@ -44,12 +35,55 @@ def get_model(input_shape,output_shape):
...
+    no_of_reviews = idx+1
+    return no_of_reviews, maxlen
+
+
+def generator(X,Y,batch_size):
+    batchX = np.zeros((batch_size,X.shape[1],80))
+    batchY = np.zeros((batch_size,1))
+    while True:
+        for i in range(batch_size):
+            idx = random.choice(range(X.shape[0]-1))
```
Comment (DennisHamann) [inline src/train.py:83]: hier bitte kein Zufallsexperiment, sondern eine Permutation nutzen. z.b shuffle()

File: src/train.py [hunk 6]
```diff
@@ -63,56 +97,23 @@ def get_model(input_shape,output_shape):
...
-    testY  = pickle.load(testYfile)
-
-    #this has do be removed once the data is complete
-
-    trainY=np.array(trainY[0:50])
-    testY=np.array(testY[0:50])
-
-    train_max_len = len(max(trainX,key=len))
-    test_max_len = len(max(testX,key=len))
+    vecsize = 80
```
Comment (DennisHamann) [inline src/train.py:100]: vecsize bitte dynamisch aus Datei laden

File: src/featurization.py (hunk 1 above)
Comment (DennisHamann) [inline src/featurization.py:213]: Dateiname soll aus input und vecsize dynamisch generiert werden.

File: src/featurization.py [hunk 7]
```diff
@@ -179,9 +207,17 @@ def getAvgFeatureVecs(reviews, model, num_features):
...
+    testXfile = csv.reader(open('data/features/80testDataVec.csv', 'rt'))
+    no_of_reviews_train, maxlen_train = get_shape(trainXfile)
+    no_of_reviews_test, maxlen_test = get_shape(testXfile)
+    shape = np.array([maxlen_train,no_of_reviews_train,maxlen_test,no_of_reviews_test])
+    np.save('data/features/shape', shape)
```
Comment (DennisHamann) [inline src/featurization.py:218]: Dateiname statisch und '.npy' fehlt.

File: src/train.py [hunk 8]
```diff
@@ -44,12 +35,55 @@ def get_model(input_shape,output_shape):
...
     #define optimizer, metric
     model.compile('adam', 'binary_crossentropy', metrics=['accuracy'])
-    #train model with given dataset
     return model
 
-#with open(output, 'w') as f:
-#    f.write('This is a trained model')
-
+def get_data(no_of_reviews_train,no_of_reviews_test,maxlen_train,maxlen_test,vecsize):
+    trainX = np.memmap('data/features/trainmapX', dtype='float', mode='r',shape=(no_of_reviews_train,maxlen_train,vecsize))
```
Comment (DennisHamann) [inline src/train.py:41]: Dateinamen sind unabhängig vom input.

File: src/train.py [hunk 9]
```diff
@@ -75,15 +75,17 @@ def get_shape(file):
...
 
 
-def generator(X,Y,batch_size):
-    batchX = np.zeros((batch_size,X.shape[1],80))
-    batchY = np.zeros((batch_size,1))
+def generator(X,Y,batch_size,vecsize,no_of_reviews,maxlen):
+    batchX = np.zeros((batch_size,maxlen,vecsize))
+    batchY = np.zeros((batch_size))
+    index=np.array(range(no_of_reviews-1))
+    random.shuffle(index)
```
Comment (DennisHamann) [inline src/train.py:82]: es sollte in der while Schleife geshuffled werden. So wird immer die gleiche Permutation verwendet

File: src/train.py [hunk 10]
```diff
@@ -21,25 +21,25 @@ def mkdir_p(path):
...
     model.compile('adam', 'binary_crossentropy', metrics=['accuracy'])
     return model
 
-def get_data(no_of_reviews_train,no_of_reviews_test,maxlen_train,maxlen_test,vecsize):
-    trainX = np.memmap('data/features/trainmapX', dtype='float', mode='r',shape=(no_of_reviews_train,maxlen_train,vecsize))
-    testX = np.memmap('data/features/testmapX', dtype='float', mode='r',shape=(no_of_reviews_test,maxlen_test,vecsize))
+def get_data(no_of_reviews_train,no_of_reviews_test,maxlen,vecsize):
+    trainX = np.memmap('data/features/trainmapX', dtype='float', mode='r',shape=(no_of_reviews_train,maxlen,vecsize))
+    testX = np.memmap('data/features/testmapX', dtype='float', mode='r',shape=(no_of_reviews_test,maxlen,vecsize))
     trainYfile = open('data/features/80trainDatalabel.pickle', 'rb')
```
Comment (DennisHamann) [inline src/train.py:43]: 80 im Dateinamen enthalten und an vielen anderen Stellen auch: bitte prüfen

File: src/featurization.py [hunk 11]
```diff
@@ -214,10 +214,11 @@ def get_shape(file):
     testXfile = csv.reader(open('data/features/80testDataVec.csv', 'rt'))
```
Comment (DennisHamann) [inline src/featurization.py:214]: 80 in Dateiname enthalten..

File: src/featurization.py (hunk 7 above)
Comment (JonathanFenske) [reply src/featurization.py]: Bei np.save wird automatisch die Endung .npy angehängt
